addon/capabilities.js

```diff
@@ -0,0 +1,41 @@
+const CAPABILITIES = {
+  requestAnimationFrame: requestAnimationFrame,
+  requestIdleCallback: requestIdleCallback,
+};
+
+/**
+ * @type {Capabilities}
+ */
+let _instance = null;
+
+class Capabilities {
+  constructor() {
+    if (_instance) {
+      throw new Error('Capabilities can only be constructed once');
+    }
+
+    this.reset();
+    this.DISABLED = true;
+  }
+
+  static get instance() {
```

Comment:
Maybe we can kill this (and the instance check in constructor), and instead just have our default export be the instance?

```js
export class Capabilities { /* stuff here */ }

export default new Capabilities();
```

Reply:
So I did that, but was seeing issues in tests where I required the module, and would get a new instance of Capabilities.

Reply:
the module should only be evaluated once, so `export default new Thing()` should only ever create a single instance

happy to dig in with you to see whats going on